pokemonterminal/wallpaper/__init__.py

```diff
@@ -16,17 +17,12 @@ def _get_adapter_classes() -> [WallpaperProvider]:
     all the implementing wallpaper adapter classes
     thanks for/adapted from https://github.com/cclauss/adapter_pattern/
     """
-    dirname = os.path.join(os.path.dirname(
-        os.path.abspath(__file__)), 'adapters')
-    adapter_classes = []
-    for file_name in sorted(os.listdir(dirname)):
-        root, ext = os.path.splitext(file_name)
-        if ext.lower() == '.py' and not root.startswith('__'):
-            module = importlib.import_module(
-                '.' + root, 'pokemonterminal.wallpaper.adapters')
+    adapter_dir = pathlib.Path(__file__).resolve().parent / 'adapters'
+    for file in adapter_dir.iterdir():
+        if file.suffix.lower() == '.py' and not file.name.startswith('__'):
+            module = importlib.import_module('.' + file.name.split('.')[0], 'pokemonterminal.wallpaper.adapters')
             for _, c in inspect.getmembers(module, _is_adapter):
-                adapter_classes.append(c)
-    return adapter_classes
+                yield c
```

Comment:
I mean looks nice and all... But is there a point other then 'it looks cooler'? Making sure we don't waste time generating a array nobody will read, would be my guess... is that it? 

And nice work with pathlib didn't know it was this much better.

And just so you know this broke the unit tests, just consume the generator over there with a `list()` constructor.